pkg/vm/createCustom.go

```diff
@@ -55,3 +78,104 @@ func loadCustomGenesis(app *application.Avalanche, genesisPath string) ([]byte,
 	genesisBytes, err := os.ReadFile(genesisPath)
 	return genesisBytes, err
 }
+
+func SetCustomVMSourceCodeFields(app *application.Avalanche, sc *models.Sidecar, customVMRepoURL string, customVMBranch string, customVMBuildScript string) error {
+	var err error
+	if customVMRepoURL != "" {
+		ux.Logger.PrintToUser("Checking source code repository URL %s", customVMRepoURL)
+		if err := prompts.ValidateURL(customVMRepoURL); err != nil {
+			ux.Logger.PrintToUser("Invalid repository url %s: %s", customVMRepoURL, err)
+			customVMRepoURL = ""
+		}
+	}
+	if customVMRepoURL == "" {
+		customVMRepoURL, err = app.Prompt.CaptureURL("Source code repository URL")
+		if err != nil {
+			return err
+		}
+	}
+	if customVMBranch != "" {
+		ux.Logger.PrintToUser("Checking branch %s", customVMBranch)
+		if err := prompts.ValidateRepoBranch(customVMRepoURL, customVMBranch); err != nil {
+			ux.Logger.PrintToUser("Invalid repository branch %s: %s", customVMBranch, err)
+			customVMBranch = ""
+		}
+	}
+	if customVMBranch == "" {
+		customVMBranch, err = app.Prompt.CaptureRepoBranch("Branch", customVMRepoURL)
+		if err != nil {
+			return err
+		}
+	}
+	if customVMBuildScript != "" {
+		ux.Logger.PrintToUser("Checking build script %s", customVMBuildScript)
+		if err := prompts.ValidateRepoFile(customVMRepoURL, customVMBranch, customVMBuildScript); err != nil {
+			ux.Logger.PrintToUser("Invalid repository build script %s: %s", customVMBuildScript, err)
+			customVMBuildScript = ""
+		}
+	}
+	if customVMBuildScript == "" {
```

Comment:
put `if customVMBuildScript == ""` first and replase `if customVMBuildScript != ""` with else to make code more clear and consistent with other if statements

Reply:
this is this way on purpose. in the param given in flags is invalid, is it set to "" and then, it is asked to the user in the following if

Reply:
so, the user is asked with prompt either if:
- no flag was provided (eg customVMBuildScript == "")
- a flag was provided but was invalid, so validate repo file fails, and after that (customVMBuildScript == "")